oxrdflib/parser.py

```diff
@@ -0,0 +1,121 @@
+import warnings
+from typing import Any, Optional, Union
+
+from rdflib import ConjunctiveGraph, Graph
+from rdflib.exceptions import ParserError
+from rdflib.parser import FileInputSource, InputSource, Parser
+
+from oxrdflib._converter import to_ox
+from oxrdflib.store import OxigraphStore
+
+__all__ = [
+    "OxigraphTurtleParser",
+    "OxigraphNTriplesParser",
+    "OxigraphRdfXmlParser",
+]
+
+
+class OxigraphParser(Parser):
+    def __init__(self):
+        pass
+
+    def parse(
+        self,
+        source: InputSource,
+        sink: Union[Graph, ConjunctiveGraph],
+        format: str,
+        encoding: Optional[str] = "utf-8",
+        **kwargs: Any,
+    ) -> None:
+        if encoding not in (None, "utf-8"):
+            raise ParserError("N3/Turtle files are always utf-8 encoded, I was passed: %s" % encoding)
+
+        if type(source) not in (FileInputSource, InputSource):
+            raise ParserError("Source must be either io(bytes) or io(str) or str or pathlib.Path")
```

Comment:
If I am not wrong, you are now using `source.getByteStream()` to get the underlying source so every implementation of `InputSource` and not only `FileInputSource` or `InputSource` should work. Am I correct?

Then it might be nice to specialize the implemention for FileInputSource in order to give to pyoxigraph only the file path and get parsing without cPython involved at all.

Reply:
Correct ! 

NP I will implement specialisation.

Reply:
Thank you! So, I guess we can remove this condition to allow other subclasses of `InputSource`? The type annotation assume that `source` is always an `InputSource`.

Reply:
Additional code added in order to handle `URLInputSource` , which require a bit of conversion of format:

ox -> rdflib 

while rest require to proper MIME type so `bulk_load` and `load` does a proper job.